src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -270,9 +277,38 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
 
             formulaResult ??= FormulaValue.NewBlank(_info.RetType);
 
-            if (!formulaResult.Type._type.Accepts(_info.RetType._type, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true))
+            var dataSourceType = formulaResult.Type._type;
+            var retType = _info.RetType._type;
+            

```

Comment:
This method is already quite large (100+ lines); consider breaking it down into smaller parts